registry/server/appRoutes/routes/RoutesRepository.ts

```diff
@@ -0,0 +1,72 @@
+import _ from 'lodash/fp';
+import { Knex } from 'knex';
+import db, { type VersionedKnex } from '../../db';
+import { Tables } from '../../db/structure';
+import { appendDigest } from '../../util/hmac';
+import { EntityTypes, VersionedRecord, VersionRow } from '../../versioning/interfaces';
+import { AppRoute, appRouteSchema, AppRouteSlot, AppRouteSlotDto, appRouteSlotSchema } from '../interfaces';
+import { extractInsertedId } from '../../util/db';
+import { prepareAppRouteToSave, prepareAppRouteSlotsToSave } from '../services/prepareAppRoute';
+import { User } from '../../../typings/User';
+
+type AppRouteDto = VersionedRecord<Omit<AppRoute, 'id'>> & AppRouteSlot;
+
+export class RoutesRepository {
+    constructor(private readonly db: VersionedKnex) {}
+
+    public getRoutesById(appRouteId: number) {
+        type QueryResult = AppRouteDto & {
+            _routeId?: number;
+        };
+        const query = this.db
+            .selectVersionedRows<QueryResult>(Tables.Routes, 'id', EntityTypes.routes, [
+                'routes.id as _routeId',
+                'routes.*',
+                'route_slots.*',
+            ])
+            .from(Tables.Routes)
+            .leftJoin('route_slots', 'route_slots.routeId', 'routes.id');
+        return query.then((appRoutes) => {
```

Comment:
Let's create techdebt item to utilise different db adapters, it can become really messy quickly